components/rollup_node/README.md

```diff
@@ -1,62 +1,109 @@
+Based on the rules of [block generation][block-gen], it is able to compute sets ("epochs") of L2 blocks as a stateless, pure function of ranges of L1 blocks.
 
-### [Sequencer Replicator][sequencer-replicator]
+While connected to the Execution Engine, it uses the Engine API to progress the L2 state as new data comes in from L1.
+The rollup driver will instruct the engine to reorganize the L2 state if it detects a difference between the local chain and remote chain as confirmed on L1,
+and provides a way for other services to subscribe to reorgs.
 
-The sequencer replicator connects directly to the sequencer. It requests pending L2 blocks which have not yet been batch submitted, and inserts them into the tip of the local EE.
+### Block Producer
 
-These blocks are marked in the EE as not yet confirmed on L1. Replicating verifiers still run a CL, which will later mark them as L1-confirmed within the EE.
+The [Block Producer][block-producer] is an optional service to run as sequencer, *within the rollup node*,
+to progress the L2 state in advance of batch submission.
 
-### [Block Producer][block-producer]
+When tasked to produce a new rollup block, the rollup node adds `PayloadAttributesV1` to the `engine_forkchoiceUpdated` 
+call to instruct the engine to start sequencing a block of pending contents from the transaction pool.
 
-The block producer service is run by the sequencer to progress the L2 state in advance of batch submission. The block producer repeatedly calls `engine_preparePayload` to get a new L2 block and then uses `engine_executePayload` to insert the resulting block into the state. When new L1 blocks appear, it passes the corresponding deposit transactions to `engine_preparePayload`.
+This execution block is then retrieved with `engine_getPayload`, to then be pushed to L1 for availability,
+and optionally directly the L2 network for faster tip updates.
 
-### [Batch Submitter][batch-submitter]
+### Batch Submitter
 
-The batch submitter takes the pending blocks which have been inserted into the sequencer's L2 state by the block producer, and sends corresponding transactions in batches to L1.
+The [Batch Submitter][batch-submitter] service takes a queue of locally produced payloads,
+and submits the corresponding transactions in batches to L1.
 
 The batch submitter encodes these transactions based on the rules of [block generation][block-gen], effectively as the reverse of the process carried out by the rollup driver. As a result:
 - Blocks previously inserted by the Block Producer into the Sequencer's EE will now appear in L1 Verifiers' EEs.
-- Blocks marked as sequencer-confirmed in both the Sequencer and in Replicating verifiers will now be marked as L1-confirmed.
+- Blocks marked as sequencer-confirmed in both the Sequencer and in Replicating verifiers will now be recognized as L1-confirmed.
+
+### P2P Interface
+
```

Comment:
No, the p2p interface is only tasked with propagating the blocks from L2 to L2. There is no mention of L1 to L2 propagation, because that is not the goal (for L1 reading, the node can just pull it from its L1 RPC connection, which is not a form of P2P, as the node is trusted). And the intention is not to replace "following L1", but to stay closer to the head of the L2, and defer verification with L1 to when the data actually gets on L1.

Reply:
I'll try clarify the trust model of L1 vs L2 more here